cosmos/providers/dbt/core/operators/docker.py

```diff
@@ -129,9 +129,13 @@ class DbtTestDockerOperator(DbtDockerBaseOperator):
 
     ui_color = "#8194E0"
 
-    def __init__(self, **kwargs) -> None:
+    def __init__(
+        self, on_warning_callback: Optional[Callable] = None, **kwargs
+    ) -> None:
         super().__init__(**kwargs)
         self.base_cmd = "test"
+        # as of now, on_warning_callback in docker executor does nothing
+        self.on_warning_callback = on_warning_callback
```

Comment:
if it doesn't do anything, can we remove it as an arg until it _is_ supported?

Reply:
if removed, it would crash the **DbtDag** and **DbtTaskGroup** methods. The reason is because the aforementioned methods can be configured to call the Test classes from all execution modes (local, docker & kubernetes) and injects the arguments into them. Since **on_warning_callback** is not defined in the test operators for docker and k8s, airflow would throw an error saying that it cant inject **on_warning_callback** into them, so I defined this "do-nothing" argument to work around the issue.